### PR TITLE
Check rmq retention only by page

### PR DESCRIPTION
**Pull request by fishpenguin**
Signed-off-by: fishpenguin <kun.yu@zilliz.com>

/kind improvement

issue: #11507

### CODE REVIEW

**Comment by congqixia** [comment]
@fishpenguin after change ack msg id to page id, will there be any compatibility issue? say some page will never be acked after upgrade to latest version use page id logic?

File: internal/util/rocksmq/server/rocksmq/rocksmq_impl.go [hunk 1]
```diff
@@ -786,24 +780,64 @@ func (rmq *rocksmq) updateAckedInfo(topicName, groupName string, ids []UniqueID,
...
+
+	fixedAckedTsKey, err := constructKey(AckedTsTitle, topicName)
+	if err != nil {
+		return err
+	}
+
+	// 3. Update acked ts and acked size for pageIDs
 	if vals, ok := rmq.consumers.Load(topicName); ok {
 		var minBeginID int64 = math.MaxInt64
 		for _, v := range vals.([]*Consumer) {
```
Comment (czs007) [inline internal/util/rocksmq/server/rocksmq/rocksmq_impl.go:839]: Could vals be empty?

Comment (fishpenguin) [reply internal/util/rocksmq/server/rocksmq/rocksmq_impl.go:839]: Can't

Comment (fishpenguin) [reply internal/util/rocksmq/server/rocksmq/rocksmq_impl.go:839]: Already do judge before.

**Pull request merged**